### PR TITLE
feat(medusa): Replace Node Redis with ioredis

### PR DESCRIPTION
**Pull request by pepijn-vanvlaanderen**
See https://github.com/medusajs/medusa/issues/5655 for more details on the rationale behind this PR.

### CODE REVIEW

File: packages/medusa/src/loaders/express.ts [hunk 1]
```diff
@@ -40,7 +40,11 @@ export default async ({ app, configModule }: Options): Promise<Express> => {
 
   if (configModule?.projectConfig?.redis_url) {
     const RedisStore = createStore(session)
-    const redisClient = redis.createClient(configModule.projectConfig.redis_url)
+    const redisClient = new Redis(configModule.projectConfig.redis_url, {
+      // Lazy connect to properly handle connection errors
+      lazyConnect: true,
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/loaders/express.ts]: `lazyConnect` will wait for a first command to be called to then connect to Redis. This should be in your options, not by default.

https://ioredis.readthedocs.io/en/latest/API/

**Review by carlos-r-l-rodrigues — CHANGES_REQUESTED** [review]

File: packages/medusa/src/loaders/express.ts (hunk 1 above)
Comment (pepijn-vanvlaanderen) [reply packages/medusa/src/loaders/express.ts]: Thanks for the review, I’ve made the adjustment.

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

Thanks for the PR.

**Review by olivermrbl — APPROVED** [review]
seems good to me :)

File: packages/medusa/src/loaders/express.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/medusa/src/loaders/express.ts]: @carlos-r-l-rodrigues this is inconsistent with how we initialize Redis in other places ... I am fine with going with this approach, but I just wanted to point it out

We typically do:
```ts
...
    const redisClient = new Redis(configModule.projectConfig.redis_url, {
      lazyConnect: true,
      ...(configModule.projectConfig.redis_options ?? {}),
    })

    try {
      await redisClient.connect()
      logger?.info(`Connection to Redis established`)
    } catch (err) {
      logger?.error(`An error occurred while connecting to Redis:${EOL} ${err}`)
    }
...
```

Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/loaders/express.ts]: Oh I see, but in this case the first command is the `.connect` itself.
The important thing is to fail if the connection is not stablished at that point.

**Pull request merged**